HPXMLtoOpenStudio/resources/data/unavailable_periods.csv

```diff
@@ -33,3 +33,5 @@ house_fan,0,1
 kitchen_fan,1,1
 bath_fan,1,1
 whole_house_fan,0,1
+battery_charging,0,1
+battery_discharging,0,0
```

Comment:
Need to think about charging/discharging with respect to both PV and grid. The simplest solution for now would be to disallow charging/discharging during a power outage even if that's not the common scenario, but it would take effort to support something different.

Reply:
I said this yesterday as well, but: I'd vote in favor of that, at least for the time being. It's basically assuming you're not capable of islanded operation [https://www.energysage.com/energy-storage/islanding-and-batteries-what-you-need-to-know/](url).

I'd certainly like to support islanded operation at some point, but that's going to require an EMS program to control charging to only happen from PV, and not the grid (so that part isn't super complicated, but is an extra step). The other issue is if you truly are in a home with solar + storage that can island and charge off PV, you're probably using some loads, at least as much as the battery can support them, which is not something we model right now. A lot of homes like this have what's called a critical loads panel [,](https://www.energysage.com/energy-storage/what-are-critical-load-panels/) and that's the only thing that gets powered.

I'd love to find some funds to be able to explicitly model trying to run off solar plus storage (or a backup generator?) during a power outage, but I think that's a big enough list it'd require some separate $$.